qutip/wigner.py

```diff
@@ -882,42 +896,64 @@ def spin_q_function(rho, theta, phi):
 
     Q = np.zeros_like(THETA, dtype=complex)
 
-    for m1 in arange(-j, j+1):
-
-        Q += binom(2*j, j+m1) * cos(THETA/2) ** (2*(j-m1)) * sin(THETA/2) ** (2*(j+m1)) * \
-             rho.data[int(j-m1), int(j-m1)]
+    for m1 in arange(-j, j + 1):
```

Comment:
P.S. I'm fine if we keep the style fixes in the  `spin_*` functions because we're rewriting those to some extent anyway.